File: ws/bonifaz/u2/WS23UnitTest/ComplexOperation/src/test/java/ec/edu/espe/complexcotroller/controller/PersonControllerTest.java

package ec.edu.espe.complexcotroller.controller;

import java.time.LocalDate;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
 *
 * @author Christian Bonifaz, Code Creators, DCCO-ESPE
 */
public class PersonControllerTest {
    
    public PersonControllerTest() {
    }

    @Test
    public void testComputeAgeInDays() {
        System.out.println("computeAgeInDays");
        LocalDate birthDate = LocalDate.parse("2001-11-13");
        long expResult = 8273;
        long result = PersonController.computeAgeInDays(birthDate);
        assertEquals(expResult, result);
    }

    @Test
    public void testComputeAgeInYears() {
        System.out.println("computeAgeInYears");
        LocalDate birthDate = LocalDate.parse("2001-11-13");
        int expResult = 22;
        int result = PersonController.computeAgeInYears(birthDate);
        assertEquals(expResult, result);
    }
    
}

File: hw/segarra/u2/HW17TestCaseSolution/ComplexOperationsJavaScript/test/PersonController.test.js
const { computeAgeInDays, computeAgeInYears } = require('../ec.edu.espe.complexoperationsjavascript.model/PersonController.js');

test('computes age in days correctly', () => {
    const birthDate = '31-01-2004';
    const result = computeAgeInDays(birthDate);
    expect(result).toBeGreaterThan(0);
});

test('computes age in years correctly', () => {
    const birthDate = '31-01-2004';
    const result = computeAgeInYears(birthDate);
    expect(result).toBe(20);
});
